chunk.js

```diff
@@ -13,24 +13,54 @@ import toInteger from './toInteger.js'
  * @returns {Array} Returns the new array of chunks.
  * @example
  *
- * chunk(['a', 'b', 'c', 'd'], 2)
+ * chunk(['a','b','c','d'], 2)
  * // => [['a', 'b'], ['c', 'd']]
  *
- * chunk(['a', 'b', 'c', 'd'], 3)
+ * chunk(['a','b','c','d'], 3)
  * // => [['a', 'b', 'c'], ['d']]
+ *
+ * chunk(['a','b','c','d'],{
+ *    size: 3,
+ *    filler: 'e'
+ * })
+ * // => [['a', 'b', 'c'], ['d', 'e', 'e']]
+ *
+ * chunk(['a','b','c','d'],{
+ *    size: 3,
+ *    filler: null
+ * })
+ * // => [['a', 'b', 'c'], ['d', null, null]]
+ *
+ * chunk(['a','b','c','d'],{
+ *    size: 3,
+ *    filler: undefined
+ * })
+ * // => [[ 'a', 'b', 'c' ], [ 'd', undefined, undefined ]]
  */
-function chunk(array, size = 1) {
-  size = Math.max(toInteger(size), 0)
+function chunk(array, options) {
+  let size
+  if (typeof options !== 'object') {
+    size = Math.max(toInteger(options), 0)
+  }
+  else {
+    size = Math.max(toInteger(options.size), 0)
+  }
   const length = array == null ? 0 : array.length
   if (!length || size < 1) {
     return []
   }
   let index = 0
   let resIndex = 0
   const result = new Array(Math.ceil(length / size))
-
   while (index < length) {
-    result[resIndex++] = slice(array, index, (index += size))
+    const arrayChunk = slice(array, index, (index += size))
```

Comment:
Maybe something like this will be a good solution too.
```
while (index < length) {
    result[resIndex++] = slice(array, index, (index += size))
}
if (result[resIndex - 1].length !== size) {
    for (let i = result[resIndex - 1].length; i < size; i++) {
        result[resIndex - 1].push(options.filler)
    }
}
```
In this case we will do check only after creating all chunks